HospitalWebAppTests/HospitalWebApplicationFactory.cs

```diff
@@ -0,0 +1,99 @@
+﻿using HealthcareBase.Model.Database;
+using HealthcareBase.Repository.HospitalResourcesRepository;
+using HealthcareBase.Repository.MedicationRepository;
+using HealthcareBase.Repository.ScheduleRepository.ProceduresRepository;
+using HealthcareBase.Repository.UsersRepository.EmployeesAndPatientsRepository;
+using HealthcareBase.Repository.UsersRepository.GeneralitiesRepository;
+using HealthcareBase.Repository.UsersRepository.SurveyRepository;
+using HealthcareBase.Repository.UsersRepository.UserAccountsRepository;
+using HealthcareBase.Repository.UsersRepository.UserFeedbackRepository;
+using HealthcareBase.Service.HospitalResourcesService.EquipmentService;
+using HealthcareBase.Service.MedicationService;
+using HealthcareBase.Service.MiscellaneousService;
+using HealthcareBase.Service.ScheduleService.ProcedureService;
+using HealthcareBase.Service.UsersService.PatientService;
+using HealthcareBase.Service.UsersService.RegistrationService;
+using HealthcareBase.Service.UsersService.UserFeedbackService;
+using HealthcareBase.Service.UsersService.UserFeedbackService.SurveyService;
+using HospitalWebAppIntegrationTests.Context;
+using Microsoft.AspNetCore.Hosting;
+using Microsoft.AspNetCore.Mvc.Testing;
+using Microsoft.Extensions.DependencyInjection;
+using Moq;
+using System;
+
+namespace HospitalWebAppIntegrationTests
+{
+    public class HospitalWebApplicationFactory<TStartup>
+        : WebApplicationFactory<TStartup> where TStartup: class
+    {
+        private string _testConnectionString;
+        private IContextFactory GetContext()
+        {
+            return new MySqlTestContextFactory(_testConnectionString);
+        }
+        private string CreateConnectionStringFromEnvironment()
+        {
+            string server = Environment.GetEnvironmentVariable("TEST_DB_PSW_SERVER");
+            string port = Environment.GetEnvironmentVariable("TEST_DB_PSW_PORT");
+            string database = Environment.GetEnvironmentVariable("TEST_DB_PSW_DATABASE");
+            string user = Environment.GetEnvironmentVariable("TEST_DB_PSW_USER");
+            string password = Environment.GetEnvironmentVariable("TEST_DB_PSW_PASSWORD");
+            if (server == null
+                || port == null
+                || database == null
+                || user == null
+                || password == null)
+                return null;
+
+            return $"server={server};port={port};database={database};user={user};password={password};";
+        }
+        protected override void ConfigureWebHost(IWebHostBuilder builder)
+        {
+            try
+            {
+                _testConnectionString = CreateConnectionStringFromEnvironment();
+            }
+            catch { }
+            if(_testConnectionString == null) _testConnectionString = "server=localhost;port=3306;database=test_clinic;user=root;password=password";
```

Comment:
I suppose we did not have the same idea about this, so the script I created does not differentiate between dev and test environments. We'll organize a meeting to discuss it.

Reply:
Dev and test environments now use the same context and the same connection.